src/Catalog/AggregateCursor.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// A <see cref="ReadCursor"/> that returns the greatest value for <see cref="Load(CancellationToken)"/> from a set of <see cref="ReadCursor"/>s based on an <see cref="IComparer{DateTime}"/>.
+    /// </summary>
+    public class AggregateCursor : ReadCursor
+    {
+        public class Least : IComparer<DateTime>
+        {
+            public int Compare(DateTime x, DateTime y)
+            {
+                return Math.Sign((y- x).Ticks);
+            }
+        }
+
+        public AggregateCursor(IEnumerable<ReadCursor> innerCursors)
+            : this(innerCursors, new Least())
+        {
+        }
+
+        public AggregateCursor(IEnumerable<ReadCursor> innerCursors, IComparer<DateTime> comparer)
+        {
+            InnerCursors = innerCursors.ToList();
+            _comparer = comparer;
+        }
+
+        public IEnumerable<ReadCursor> InnerCursors { get; private set; }
+
+        private IComparer<DateTime> _comparer;
+
+        public override async Task Load(CancellationToken cancellationToken)
+        {
+            await Task.WhenAll(InnerCursors.Select(c => c.Load(cancellationToken)));
+
+            /// Choose the greatest value from the <see cref="InnerCursors"/> as the value for the <see cref="AggregateCursor"/> based on the <see cref="IComparer{DateTime}"/>.
```

Comment:
I'm using `//` instead of `///` because it allows me to make direct XML doc references to the code that change with changes to the code.